AEPMessaging/Sources/Messaging+EdgeEvents.swift

```diff
@@ -269,74 +275,66 @@ extension Messaging {
     ///   - interaction: a `String` describing the interaction
     ///   - message: the `Message` for which the interaction should be recorded
     func sendPropositionInteraction(withEventType eventType: MessagingEdgeEventType, andInteraction interaction: String?, forMessage message: Message) {
-        guard let propInfo = message.propositionInfo,
-              var scopeDetails = propInfo.scopeDetails.asDictionary() else {
+        guard let propInfo = message.propositionInfo, !propInfo.scopeDetails.isEmpty else {
             Log.debug(label: MessagingConstants.LOG_TAG, "Unable to send a proposition interaction - `scopeDetails` were not found for message (\(message.id)).")
             return
         }
 
-        // need to add cjmXdm object to characteristics if this is an interact event
-        if eventType == .inappInteract {
-            guard var characteristics = scopeDetails[MessagingConstants.XDM.IAM.Key.CHARACTERISTICS] as? [String: Any] else {
-                Log.debug(label: MessagingConstants.LOG_TAG, "Unable to send a proposition interaction - `characteristics` were not found for message (\(message.id)).")
-                return
-            }
-
-            let inappMessageTracking = [
-                MessagingConstants.XDM.IAM.Key.ACTION: interaction
-            ]
-            let cjmXdm = [
-                MessagingConstants.XDM.IAM.Key.IN_APP_MESSAGE_TRACKING: inappMessageTracking
-            ]
-            let encoder = JSONEncoder()
-            guard let cjmXdmData = try? encoder.encode(cjmXdm) else { return }
-            let cjmXdmString = String(data: cjmXdmData, encoding: .utf8)
-            characteristics[MessagingConstants.XDM.IAM.Key.CJM_XDM] = cjmXdmString
-            scopeDetails[MessagingConstants.XDM.IAM.Key.CHARACTERISTICS] = characteristics
-        }
-
         let propositions: [[String: Any]] = [
             [
                 MessagingConstants.XDM.IAM.Key.ID: propInfo.id,
                 MessagingConstants.XDM.IAM.Key.SCOPE: propInfo.scope,
-                MessagingConstants.XDM.IAM.Key.SCOPE_DETAILS: scopeDetails
+                MessagingConstants.XDM.IAM.Key.SCOPE_DETAILS: propInfo.scopeDetails
             ]
         ]
+        
         let propositionEventType: [String: Int] = [
             eventType.propositionEventType: 1
         ]
-        let decisioning: [String: Any] = [
+        
+        var decisioning: [String: Any] = [
             MessagingConstants.XDM.IAM.Key.PROPOSITION_EVENT_TYPE: propositionEventType,
             MessagingConstants.XDM.IAM.Key.PROPOSITIONS: propositions
         ]
+        
+        // only add `propositionAction` data if this is an interact event
+        if eventType == .inappInteract {
+            let propositionAction: [String: String] = [
+                MessagingConstants.XDM.IAM.Key.ID: interaction ?? "",
+                MessagingConstants.XDM.IAM.Key.LABEL: interaction ?? ""
```

Comment:
id and label should both contain the interaction?

Reply:
for now, yes. 

in the future we may allow there to be a separation of these values (if it's supported in the authoring ui)